lib/models/watch-detector.js

```diff
@@ -160,32 +155,31 @@ WatchDetector.prototype.findBestWatcherOption = function(options) {
     preference = this.checkWatchman();
   }
 
-  return Promise.resolve(preference).then(function(bestOption) {
-    if (bestOption.watcher === NODE) {
-      // although up to this point, we may believe Node is the best watcher
-      // this may not be true because:
-      // * not all platforms that run node, support node.watch
-      // * not all file systems support node watch
-      //
-      return this.testIfNodeWatcherAppearsToWork().then(function(appearsToWork) {
-        return new WatchPreference(appearsToWork ? NODE : POLLING);
-      });
-    } else {
-      return bestOption;
-    }
-  }.bind(this)).then(function(actual) {
-    debug.info('foundBestWatcherOption, preference was: %o, actual: %o', options, actual);
-    if (actual /* if no preference was initial set, don't bother informing the user */ &&
-      original /* if no original was set, the fallback is expected */ &&
-      actual.watcher !== original &&
-
-      // events represents either watcherman or node, but no preference
-      !(original === EVENTS && (actual.watcher === WATCHMAN || actual.watcher === NODE))) {
-      // if there was an initial preference, but we had to fall back inform the user.
-      this.ui.writeLine('was unable to use: "' + original + '", fell back to: "' + actual.watcher + '"')
-    }
-    return actual;
-  }.bind(this));
+  var bestOption = preference;
+  var actual;
+
+  if (bestOption.watcher === NODE) {
+    // although up to this point, we may believe Node is the best watcher
+    // this may not be true because:
+    // * not all platforms that run node, support node.watch
+    // * not all file systems support node watch
+    //
+    var appearsToWork = this.testIfNodeWatcherAppearsToWork();
+    actual = new WatchPreference(appearsToWork ? NODE : POLLING);
+  } else {
+    actual = bestOption;
+  }
+  debug.info('foundBestWatcherOption, preference was: %o, actual: %o', options, actual);
+  if (actual /* if no preference was initial set, don't bother informing the user */ &&
```

Comment:
This whole thing is bonkers. Can we create intermediate variables along the way?